File: rc-application/src/main/java/com/transkript/reportcard/business/service/TermService.java
package com.transkript.reportcard.business.service;


import com.transkript.reportcard.api.dto.TermDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.Term;

import java.util.List;

public interface TermService {
    EntityResponse addTerm(TermDto termDto);

    List<TermDto> getTerms();

    TermDto getTerm(Long id);

    Term getTermEntity(Long id);

    EntityResponse updateTerm(Long id, TermDto termDto);

    String deleteTerm(Long id);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/interf/TermService.kt
package com.transkript.reportcard.business.service.interf

import com.transkript.reportcard.api.dto.TermDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.Term

interface TermService {
    fun addTerm(termDto: TermDto?): EntityResponse?
    val terms: List<TermDto?>?
    fun getTerm(id: Long?): TermDto?
    fun getTermEntity(id: Long?): Term?
    fun updateTerm(id: Long?, termDto: TermDto?): EntityResponse?
    fun deleteTerm(id: Long?): String?
}
